elements/lisk-codec/src/string.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { writeVarInt } from './varint';
+
+interface SchemaProperty {
+	readonly dataType: string;
+}
+
+export const writeString = (value: string, _schema: SchemaProperty): Buffer =>
+	Buffer.concat([
+		Buffer.from('0a', 'hex'),
+		writeVarInt(value.length, { dataType: 'uint32' }),
```

Comment:
This needs to be the length of the buffer not the value itself (it might or not be the same)